src/internal/coredb/projects_test.go

```diff
@@ -0,0 +1,153 @@
+//go:build unit_test
+
+package coredb
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gogo/protobuf/types"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/clusterstate"
+	"github.com/pachyderm/pachyderm/v2/src/internal/dockertestenv"
+	"github.com/pachyderm/pachyderm/v2/src/internal/migrations"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+	"github.com/pachyderm/pachyderm/v2/src/internal/require"
+	"github.com/pachyderm/pachyderm/v2/src/internal/stream"
+	"github.com/pachyderm/pachyderm/v2/src/internal/testetcd"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+)
+
+const (
+	testProj     = "testproj"
+	testProjDesc = "this is a test project"
+)
+
+func TestCreateProject(t *testing.T) {
+	ctx := pctx.TestContext(t)
+	db := dockertestenv.NewTestDB(t)
+	tx, err := db.BeginTxx(ctx, nil)
+	require.NoError(t, err, "should be able to create a tx")
+	defer tx.Rollback()
+	migrationEnv := migrations.Env{EtcdClient: testetcd.NewEnv(ctx, t).EtcdClient}
+	require.NoError(t, migrations.ApplyMigrations(ctx, db, migrationEnv, clusterstate.DesiredClusterState), "should be able to set up tables")
+	createInfo := &pfs.ProjectInfo{Project: &pfs.Project{Name: testProj}, Description: testProjDesc}
+	require.NoError(t, CreateProject(ctx, tx, createInfo), "should be able to create project")
+	getInfo, err := GetProjectByName(ctx, tx, testProj)
+	require.NoError(t, err, "should be able to get a project")
+	require.Equal(t, createInfo.Project.Name, getInfo.Project.Name)
+	require.Equal(t, createInfo.Description, getInfo.Description)
```

Comment:
It would probably be good to try creating a project with the same name here and seeing it fail because of the unique constraint.